src/Microsoft.AspNetCore.HttpsPolicy/HstsMiddleware.cs

```diff
@@ -59,10 +64,25 @@ public Task Invoke(HttpContext context)
         {
             if (context.Request.IsHttps)
             {
-                context.Response.Headers[HeaderNames.StrictTransportSecurity] = _strictTransportSecurityValue;
+                if (!IsDomainBlocked(context.Request.Host.Host))
+                {
+                    context.Response.Headers[HeaderNames.StrictTransportSecurity] = _strictTransportSecurityValue;
+                }
             }
 
             return  _next(context);
         }
+
+        private bool IsDomainBlocked(string host)
+        {
+            for (var i = 0; i < _excludedDomains.Count; i++)
+            {
+                if (host.Equals(_excludedDomains[i], StringComparison.OrdinalIgnoreCase))
```

Comment:
Use `string.Equals(host, ...);` host may be null.
https://github.com/aspnet/HttpAbstractions/blob/cb6cf16b7a67465afac8ac3a1333bac51d16f07f/src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs#L279